src/scripts/batchRequest.ts

```diff
@@ -0,0 +1,68 @@
+import {BatchRequestOperation} from "./structuredTypes";
+
+/**
+ * The BATCH API allows a user to execute multiple OneNoteApi actions in a single HTTP request.
+ * For example, sending two PATCHES in the same HTTP request
+ * To use, construct a new BatchRequest and then pass in an object that adheres to the BatchRequestOperation interface into
+ * 	BatchRequest::addOperation(...). Once the request is built, send it using OneNoteApi::sendBatchRequest(...)
+ */
+export class BatchRequest {
+	private operations: BatchRequestOperation[];
+	private boundaryName: string;
+	private requestBody: string;
+
+	constructor() {
+		this.operations = [];
+		this.boundaryName = "batch_" + Math.floor(Math.random() * 1000);
+	}
+
+	public addOperation(op: BatchRequestOperation): void {
+		this.operations.push(op);
+	}
+
+	public getOperation(index: number): BatchRequestOperation {
+		return this.operations[index];
+	}
+
+	public getOperations(): BatchRequestOperation[] {
+		return this.operations;
+	}
+
+	public getNumOperations(): number {
+		return this.operations.length;
+	}
+
+	public getRequestBody(): string {
+		return this.convertOperationsToHttpRequestBody();
+	}
+
+	private convertOperationsToHttpRequestBody(): string {
```

Comment:
getRequestBody() simply returns convertOperationsToHttpRequestBody(). You could remove the private method and move its body into getRequestBody()